kv/kvserver: TestLeaseExpirationBasedDrainTransferWithExtension failed

[(kv/kvserver).TestLeaseExpirationBasedDrainTransferWithExtension failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2495086&tab=buildLog) on [master@aa69b42032f89d1bb777986ff7a8d4f753b71d83](https://github.com/cockroachdb/cockroach/commits/aa69b42032f89d1bb777986ff7a8d4f753b71d83):

```
=== RUN   TestLeaseExpirationBasedDrainTransferWithExtension
    test_log_scope.go:72: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLeaseExpirationBasedDrainTransferWithExtension150952326
    test_log_scope.go:73: use -show-logs to present logs inline
    client_replica_test.go:580: attempting to send snapshot that does not contain the recipient as a replica; snapshot type: LEARNER_INITIAL, recipient: s(n2,s2):2LEARNER, desc: r1:/M{in-ax} [(n1,s1):1, next=3, gen=2]
    panic.go:617: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLeaseExpirationBasedDrainTransferWithExtension150952326
I201204 06:17:21.700537 550139 kv/kvserver/replica_destroy.go:176  [s2,r1/2:{-}] removed uninitialized range in 0ms [clear=0ms commit=0ms]
I201204 06:17:21.700615 550139 kv/kvserver/store_remove_replica.go:229  [s2,r1/2:{-}] removing uninitialized replica [n2,s2,r1/2:{-}]
--- FAIL: TestLeaseExpirationBasedDrainTransferWithExtension (0.16s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestLeaseExpirationBasedDrainTransferWithExtension PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestLeaseExpirationBasedDrainTransferWithExtension.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/kv/kvserver/replica_command.go
+++ b/pkg/kv/kvserver/replica_command.go
@@ -2031,10 +2031,9 @@ func (r *Replica) sendSnapshot(
 	// the leaseholder and we haven't yet applied the configuration change that's
 	// adding the recipient to the range.
 	if _, ok := snap.State.Desc.GetReplicaDescriptor(recipient.StoreID); !ok {
-		return errors.Newf(
-			"attempting to send snapshot that does not contain the recipient as a replica; "+
-				"snapshot type: %s, recipient: s%d, desc: %s",
-			snapType, recipient, snap.State.Desc)
+		err := errors.Newf("attempting to send snapshot that does not contain the recipient as a replica; "+
+			"snapshot type: %s, recipient: s%d, desc: %s", snapType, recipient, snap.State.Desc)
+		return errors.Mark(err, errMarkSnapshotError)
 	}
 
 	sender, err := r.GetReplicaDescriptor()

